src/main/java/com/puppycrawl/tools/checkstyle/JavaAstVisitor.java

```diff
@@ -1769,12 +1772,15 @@ public DetailAstImpl visitArrayCreatorRest(JavaLanguageParser.ArrayCreatorRestCo
             if (ctx.children.get(i) == ctx.RBRACK()) {
                 arrayDeclarator.addChild(create(ctx.RBRACK()));
             }
-            else if (ctx.children.get(i) == ctx.expression()) {
-                // Handle '[8]', etc.
-                arrayDeclarator.addChild(visit(ctx.expression()));
-            }
             else {
-                addLastSibling(arrayDeclarator, visit(ctx.children.get(i)));
+                final JavaLanguageParser.ExpressionContext expression = ctx.expression();
```

Comment:
Can we just extract `final ParseTree child = ctx.children.get(i);` before any logic in the for loop, and use `child` in all conditions? I would like to avoid nested if/else where possible.

Reply:
done

Reply:
Resolved.